### PR TITLE
Make qualification test flexible about what configs to test

### PR DESCRIPTION
**Pull request by bmerry**
- Allow a maximum number of antennas to be set (and test a range of interesting antenna counts up to the maximum)
- Allow the channelisations and bands to be chosen
- Add definitions for S-band (5 variants for the different sub-bands)

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [ ] If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-695.

### CODE REVIEW

**Comment by bmerry** [comment]
Here's a preliminary example of a qualification report. I had to hack things a bit because there was an error in the S-band definition at the time and so those tests failed. I'll update when I have a better one.

[report-NGC-695.pdf](https://github.com/ska-sa/katgpucbf/files/10697715/report-NGC-695.pdf)

**Comment by bmerry** [comment]
Here is a more complete report: [report-NGC-695.pdf](https://github.com/ska-sa/katgpucbf/files/10704337/report-NGC-695.pdf)

File: qualification/conftest.py [hunk 1]
```diff
@@ -111,6 +116,24 @@ def pytest_report_collectionfinish(config: pytest.Config) -> None:  # noqa: D103
...
+            values = [min(max_antennas, DEFAULT_ANTENNAS)]
+        values = [value for value in values if value <= max_antennas]
+        metafunc.parametrize("n_antennas", values, indirect=True)
+    if "band" in metafunc.fixturenames:
+        metafunc.parametrize("band", metafunc.config.getini("bands"), indirect=True)
+    if "n_channels" in metafunc.fixturenames:
+        values = [int(value) for value in metafunc.config.getini("channels")]
+        metafunc.parametrize("n_channels", values, indirect=True)
+
+
```
Comment (james-smith-za) [inline qualification/conftest.py:136]: Nice.

**Review by james-smith-za — COMMENTED** [review]
This seems to have broken your previous efforts to marshall similar configs into a common correlator with a reset in-between tests. So for 252 tests, 252 individual correlators were started.

I don't have a problem with this in principle, but I'm not entirely sure whether you wanted / expected that to happen.

**Comment by bmerry** [comment]
> This seems to have broken your previous efforts to marshall similar configs into a common correlator with a reset in-between tests. So for 252 tests, 252 individual correlators were started (resulting in ~100 pages of correlator configs in the report).
> 
> I don't have a problem with this in principle, but I'm not entirely sure whether you wanted / expected that to happen.

Drat. That probably slows down the testing quite a lot. I'll need to revisit to see why that didn't work. Hopefully it's just a matter of explicitly providing scope when parametrizing.

**Comment by bmerry** [comment]
> This seems to have broken your previous efforts to marshall similar configs into a common correlator with a reset in-between tests. So for 252 tests, 252 individual correlators were started (resulting in ~100 pages of correlator configs in the report).

Fixed now, but running a new qualification test to confirm.

**Comment by bmerry** [comment]
New qualification test report: [report-NGC-695.pdf](https://github.com/ska-sa/katgpucbf/files/10706607/report-NGC-695.pdf). It is more than twice as fast and uses a more reasonable number of correlator configs. I'll file a separate ticket for the UHF failures.

**Review by james-smith-za — APPROVED** [review]

**Pull request merged**